Various frontend improvements

rd3k opened the pull request:
Optimise shadow image (25.7% filesize reduction), fix chmod in README, HTML5 doctype (why was it "loose"?), make all JS external, concatenate and minified.

rd3k committed:
Various frontend improvements

Optimise shadow image, fix chmod in README, HTML5 doctype, make all JS
external, concat+minify JS

Pezmc commented:
Looks great to me, not sure why it was loose. I did the frontend in a rush perhaps!

Pezmc commented on index.html:
```diff
@@ -1,10 +1,9 @@
 <meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
 <title>CS Timetable to iCal</title>
 <script src="//ajax.googleapis.com/ajax/libs/jquery/1.10.2/jquery.min.js"></script>
-<script src="js/jquery.steps.min.js"></script>
```
Where was this moved to? (Reviewing on a phone)

rd3k replied:
It's been concatenated with main.js into main.min.js. I've got this automated in CodeKit, but perhaps there should be a "build script" for if main.js ever changes.

Pezmc replied:
That's probably a good idea!

## 

07845 594224
Sent from a mobile device

rd3k committed:
More frontend tidying

Add “tick all” and “clear” buttons to course list, setting height of
course list so it becomes a scrollable area with “submit” button is
always visible at the bottom (looks better when you show all courses
for a year)

rd3k commented on templates/page4.twig:
```diff
@@ -1,6 +1,10 @@
+<input id="tickall" type="button" class="clean-gray" value="Tick all"/>
+<input id="clear" type="button" class="clean-gray" value="Clear"/>
```
Good idea?

Pezmc replied:
Yep great! Can't merge till I have my laptop though
